include/nuttx/sched.h

```diff
@@ -678,6 +677,10 @@ struct task_tcb_s
 
   struct tcb_s cmn;                      /* Common TCB fields               */
 
+  /* Task Group *************************************************************/
+
+  struct task_group_s group;            /* Pointer to shared task group data */
```

Comment:
but, main thread may exist before other thread, the used-after-free will happen.

Reply:
**Task** will kill all **pthread**s in the group before exiting. Will the **task** exit early than **pthread**?

```
#0  nxsched_release_tcb (tcb=0x409f04 <nxtask_exithook+96>, ttype=255 '\377') at sched/sched_releasetcb.c:99
#1  0x000000000040a0e5 in nxtask_terminate (pid=5) at task/task_terminate.c:122
#2  0x00000000004087c7 in pthread_cancel (thread=5) at pthread/pthread_cancel.c:110
#3  0x000000000040867b in group_cancel_children_handler (pid=5, arg=0x4) at group/group_killchildren.c:113
#4  0x000000000040a342 in group_foreachchild (group=0x7ffff3db53e0, handler=0x408613 <group_cancel_children_handler>, arg=0x4) at group/group_foreachchild.c:70
#5  0x000000000040870e in group_kill_children (tcb=0x7ffff3db52a0) at group/group_killchildren.c:212
#6  0x0000000000407adb in _exit (status=0) at task/exit.c:67
#7  0x0000000000404222 in exit (status=0) at stdlib/lib_exit.c:126
#8  0x000000000040dffe in nxtask_startup (entrypt=0x444546 <hello_main>, argc=1, argv=0x7ffff3db58c0) at sched/task_startup.c:70
#9  0x00000000004076b4 in nxtask_start () at task/task_start.c:134
#10 0x0000000000413081 in pre_start () at sim/sim_initialstate.c:52
```

Reply:
exit terminate the whole proccess, but pthread_exit terminate the current thread. You can reproduce the case by:

1. Create a work thread and call printf in a loop
2. Exit the main thread by pthread_exit

Reply:
Got, I have a update which delayed the release of the parent tcb, the parent tcb will be released after all child threads exit.